distribution/docker/alpine/Dockerfile

```diff
@@ -1,20 +1,53 @@
-FROM docker.io/alpine:3.14 AS builder
+ARG ALPINE_IMAGE=docker.io/alpine:3.15
 
-WORKDIR /vector
+#-------------------------------------------------------------------------------
+FROM ${ALPINE_IMAGE} AS staging
 
-COPY vector-*-unknown-linux-musl*.tar.gz ./
-RUN tar -xvf vector-0*-"$(cat /etc/apk/arch)"-unknown-linux-musl*.tar.gz --strip-components=2
+# Set a well-known working directory for staging
+WORKDIR /staging
 
-RUN mkdir -p /var/lib/vector
+# Prepare a staging root filesystem
+RUN mkdir -p \
+  rootfs/etc/vector \
+  rootfs/usr/bin \
+  rootfs/var/lib/vector
 
-FROM docker.io/alpine:3.14
-RUN apk --no-cache add ca-certificates tzdata
+COPY docker-entrypoint.sh rootfs/usr/bin/docker-entrypoint.sh
+COPY --from=docker.io/tianon/gosu:latest gosu rootfs/usr/bin/gosu
```

Comment:
Yeah, unfortunately they don't seem to _tag_ the images in a way that's not "latest". I can look to swap it out for the SHA, it's also been about a year since the last publish so it doesn't seem like there's too much risk of things churning under us.

Reply:
Ah how unfortunate :(

I think your points are all valid to deliberately use `latest` then. A pity they don't tag properly.
As you say, specially a burden if different hashes needs to be maintained for different architectures.

Reply:
We can look into options in another issue, maybe switch to a tool that provides better stability - maybe publish ourselves - or just deal with having the chunky Dockerfile including:

```dockerfile
RUN set -eux; \
	apk add --no-cache --virtual .fetch-deps dpkg gnupg; \
	dpkgArch="$(dpkg --print-architecture | awk -F- '{ print $NF }')"; \
	wget -O /usr/local/bin/gosu.asc "https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$dpkgArch.asc"; \
	wget -O /usr/local/bin/gosu "https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$dpkgArch"; \
	GNUPGHOME="$(mktemp -d)"; export GNUPGHOME; \
	gpg --batch --keyserver hkps://keys.openpgp.org --recv-keys B42F6819007F00F88E364FD4036A9C25BF357DD4; \
	gpg --batch --verify /usr/local/bin/gosu.asc /usr/local/bin/gosu; \
	gpgconf --kill all; \
	rm -rf "$GNUPGHOME"; unset GNUPGHOME; \
	apk del --no-network .fetch-deps; \
	chmod +x /usr/local/bin/gosu;
```

Reply:
Ah that is not a bad idea indeed. Instead of copying from an existing image (that can actually disappear out of the blue due to DockerHub's recent policies of deleting unused images), you install it manually from GitHub.

I do this very often actually in our Docker images. Of course, our Dockerfiles do not look so chunky because we don't need to maintain multiple architectures :)

Anyway, I think the final decision depends on how much guarantee you need/want for reproducibility of your images.

Reply:
Dropped the usage of `gosu` for now.